### ISSUE TITLE
Unable to create Linked struct in go repl

### ISSUE THREAD

**Issue by ihgann** [gopherdata/gophernotes#113]
If I create a code block for `Go` that simulates a type of Linked object, I receive an error:

```go
type Node struct {
    Next *Node
}
```

Response:

```sh
repl.go:2:11: undefined identifier: Node
```

Unfortunately this limits quite a few data structures from being able to be presented.

**Comment by SpencerPark** [comment]
I just tried that example while on #105 and things seemed to work fine so I think this is a problem in the old interpreter which will be updated when #105 is merged. There are a lot of new goodies that @cosmos72 has added to gomacro since the vendored copy currently in gophernotes .

**Comment by cosmos72** [comment]
Confirmed, recent versions of the interpreter support recursive types as the example above.

I re-opened a pull request to update the vendored copy of gomacro interpreter, hopefully it will be merged soon (see issue #112)

**Comment by cosmos72** [comment]
Working further on it.

The type declaration is easy to emulate, and gophernotes now supports it too, see commit 2ad243b439d9b5397c137acd51a1f47bd00b71e0.

Unluckily there is no way to implement it exactly with the current approach: `reflect.StructOf()` cannot create recursive types. As a consequence, the interpreter needs a lot of tricks at runtime because of the emulation, and I am not convinced I caught all the cases.

**Comment by cosmos72** [comment]
In alternative, if I do not find problems with the (not yet merged) gomacro patch, I will merge it into gophernotes sometimes next week

**Comment by cosmos72** [comment]
Fix applied in commit 48b1ccc7d1b0bb07c278e2e66016b6d2507cea78

Please report if it works or you still get errors.
